### ISSUE TITLE
Unable to show tabs and menubar

### ISSUE THREAD

**Issue by Reinier1994** [sumatrapdfreader/sumatrapdf#549]
If I disable tabs in the preferences screen, the menubar will show. The command codes in notepad also show the value "true" for showing menubar. Only when tabs are disabled, the menubar will show and otherwise it's behind the hamburger menu.

Is it possible that tabs AND menubar can be shown at the same time?

**Comment by GitHubRulesOK** [comment]
note this seems to be a "buggy" inconsistant feature since  I can get both sometimes 
1) set menubar to true, and open a pdf, often if tabs option was set on then menubar does not show
2) switch off option to use tabs and as stated by Reinier1994 the menubar will show instead of tabs
3) reset tabs to checked on in options
4) open another file and now you have both tabs and menubar (AS DESIRED) for the session
5) close files and recall SumatraPDF now the menubar is not visible (not as expected)

**Comment by mzso** [comment]
[**@Reinier1994**](https://github.com/Reinier1994) commented on [2016. jún. 8. 12:21 CEST](https://github.com/sumatrapdfreader/sumatrapdf/issues/549#issue-159129425 "2016-06-08T10:21:50Z - Replied by Github Reply Comments"):
> Is it possible that tabs AND menubar can be shown at the same time?

This restriction is in fact a "feature". No reason/rationale was ever given as for why can't you enable both anymore.
No matter how I look at it, it makes zero sense to not allow both the menubar and tabs to appear.

So I either you have the trendy and horrible hamburger menu, or you don't have tabs which is worse. So the best you can do is dig up an old dev build between 2.5.2 and 3.0

Such as this:
[SumatraPDF_tabs+menu.zip](https://github.com/sumatrapdfreader/sumatrapdf/files/511297/SumatraPDF_tabs.menu.zip)

**Comment by GitHubRulesOK** [comment]
@kjk note the above attachment includes an out of date developer release

As I previously mentioned it is possible for the user to achieve the same with a current release, but would be even better if it was an allowable setting via UseMenubar = true  and UseTabs= true
![image](https://user-images.githubusercontent.com/12754913/69435254-1848ad80-0d37-11ea-9c10-9ca8ef1bf2b2.png)

**Comment by herrfrei** [comment]
Tried it right now and switching both to true does not work (Version 3.3.3). Only the tabs are shwon, menu is hidden.

**Comment by GitHubRulesOK** [comment]
@herrfrei that is correct behavior
you can choose either method using the "newer" tabs interface by design required replacing the older non tabs header (menubar)
the principle is the canvas should be as large as possible on a phone/tablet or other small device and interface minimal, some users want neither just like MuPDF no tabs (or menu visible) others want multiple lines of icons filling half a small screen.

**Comment by mzso** [comment]
[**@GitHubRulesOK**](https://github.com/GitHubRulesOK) commented on [2022. máj. 14. 12:41 CEST](https://github.com/sumatrapdfreader/sumatrapdf/issues/549#issuecomment-1126688998 "2022-05-14T10:41:46Z - Replied by Github Reply Comments"):
> [@herrfrei](https://github.com/herrfrei) that is correct behavior  
> you can choose either method using the "newer" tabs interface by design required replacing the older non tabs header (menu bar)
> 
> The principle is the canvas should be as large as possible on a phone/tablet or other small device and interface minimal, some users want neither just like MuPDF no tabs (nor menu visible) others want multiple lines of icons filling half a small screen.
> 
> [![image](https://user-images.githubusercontent.com/12754913/168422749-c99290d4-665a-4602-8a79-b44968fd06fe.png)](https://user-images.githubusercontent.com/12754913/168422749-c99290d4-665a-4602-8a79-b44968fd06fe.png)

Sumatra is not a mobile app, it never was.

**Comment by GitHubRulesOK** [comment]
@mzso 
Agreed I don't use my old widows phone anymore for pdf reading (my fingers are too chubby) and thus I use a 600 pixel netbook much of the time or 768  pixel touch tablet as just wide enough for older icon set. call me a luddite but windows 7 worked fast enough on those but am currently using non touch high speed hi-res laptop for working not leisure.

**Comment by mzso** [comment]
@GitHubRulesOK 
You're contradicting yourself a lot. First you claim it's correct, than the you say you use Sumatra with larger displays.
Anyway, I think it's quite sure, since it's a windows program that it's mostly running on PCs and laptops.

**Comment by GitHubRulesOK** [comment]
@mzso I am <strike>Devils</strike> Users Advocate,  
As supporter I can argue for both sides, on the one hand the Application Ethos was Screen Viewer with minimal graphics UI with as many features as possible available via a keyboard or mouse. Touch screen/pad features are limited by lack of a programming team. 
My "argument" is that there are both ends of scale of users to be catered for not just those that like me can afford hi-definition screen, but need a lightweight viewer for their Raspberry PI with Windows or Whine

**Comment by meteorquake** [comment]
I would love to see this resolved, as I myself like to have the menu bar always present (I use a large screen and use the File - (exec) commands a lot) and at times also want tabs...
* If I open with menu and no tabs I get the menu, and if without closing I change tabs to true I get the desired outcome of tabs + menu, but when then closed it loads up with just tabs. This shows there's no fundamental issue with having both.
* Arguments about saving screen space don't hold because *the user can themself choose in options to set one or both as they wish*; there's no reason Tabs and Menubar should reduce to Tabs only when the user can just chose Tabs alone if they desire that.
David

**Comment by GitHubRulesOK** [comment]
@meteorquake  actually seems to resolve the tabs background issue when both active :-) but dependant on users screen layout can be over bearing.
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/a042c34e-49c3-418b-8838-3314840455a8) 
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/11d5c257-c4c7-4d85-936f-a445ac0c6966)

**Comment by mzso** [comment]
[**@GitHubRulesOK**](https://github.com/GitHubRulesOK) commented on [Jun 18, 2024, 2:48 PM GMT+2](https://github.com/sumatrapdfreader/sumatrapdf/issues/549#issuecomment-2176021156 "2024-06-18T12:48:12Z - Replied by Github Reply Comments"):
> @meteorquake  actually seems to resolve the tabs background issue when both active :-) but dependant on users screen layout can be over bearing.> 
> _Originally posted by @GitHubRulesOK in https://github.com/sumatrapdfreader/sumatrapdf/issues/549#issuecomment-2176021156_

What seems to resolve?

**Comment by GitHubRulesOK** [comment]
@mzso  you are right there is no difference in themed backgrounds, so deleted my off topic remark

**Comment by ahakkar** [comment]
I am using version 3.5.2 on Windows. Has there been any progress on this issue during last eight years? I still fail to intuitively figure out how to show the File, View, Go to. etc. at the same time as tabs. If this is somehow possible it is behind some truly obfuscated process which leads to bad user experience.

**Comment by GitHubRulesOK** [comment]
@ahakkar 
no it is "unintentional" to have both
but can be triiggered so if seen it is considered a "bug" by some or a "feature" by others

**Comment by ahakkar** [comment]
So perhaps this should be closed as "won't fix" or something like that? 😊

**Comment by GitHubRulesOK** [comment]
There are many abilities that are not part of main application and thus use of SumatraPDF in Browsers or browsing with SumatraPDF features function both ways but are thus "wont fix" but not listed as such just simply "historic functions.
![image](https://github.com/user-attachments/assets/8a451251-2303-4841-8fc9-2607de642f5b)

**Comment by onlinestatements** [comment]
> There are many abilities that are not part of main application and thus use of SumatraPDF in Browsers or browsing with SumatraPDF features function both ways but are thus "wont fix" but not listed as such just simply "historic" functions.

I have installed the latest pre-release 3.6.16779 but how do i get it to show the tabs and the file menu bar at the same time like you did?

**Comment by GitHubRulesOK** [comment]
@onlinestatements it WAS NOT a "working feature" (but is now in 3.7.## but old behaviour read my comment https://github.com/sumatrapdfreader/sumatrapdf/issues/549#issuecomment-226971909
![image](https://github.com/user-attachments/assets/377ec66e-cce0-4e1f-a167-1bbca99fc03e)

**Comment by kjk** [comment]
It's now possible in latest pre-release

**Comment by mzso** [comment]
@kjk 
> It's now possible in latest pre-release

It has some issues though. It doesn't get applied unless I change the window size after it. When flipping the option back and forth sometimes it was partially applied I think, I ended up without both a menubar and a menu button, and once the toolbar was partially clipped off as well.

**Comment by kjk** [comment]
improved
